r/vignettes/developing.Rmd

```diff
@@ -483,60 +434,156 @@ variables or other settings:
+
 * Tests that require allocating >2GB of memory to test Large types are disabled
   unless `ARROW_LARGE_MEMORY_TESTS=true`
+  
 * Integration tests against a real S3 bucket are disabled unless credentials
   are set in `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`; these are available
   on request
+  
 * S3 tests using [MinIO](https://min.io/) locally are enabled if the
   `minio server` process is found running. If you're running MinIO with custom
   settings, you can set `MINIO_ACCESS_KEY`, `MINIO_SECRET_KEY`, and
   `MINIO_PORT` to override the defaults.
 
-## Github workflows
+## Running checks
+
+You can run package checks by using `devtools::check()` and check test coverage with `covr::package_coverage()`.
+
+```r
+# All package checks
+devtools::check()
+
+# See test coverage statistics
+covr::report()
+covr::package_coverage()
+
+```
+
+For full package validation, you can run the following commands from a terminal.
+
+```
+R CMD build .
+R CMD check arrow_*.tar.gz --as-cran
+```
+
+
+## Running additional CI checks
 
 On a pull request, there are some actions you can trigger by commenting on the PR. We have additional CI checks that run nightly and can be requested on demand using an internal tool called [crosssbow](https://arrow.apache.org/docs/developers/crossbow.html). A few important GitHub comment commands include:
 
-* `@github-actions crossbow submit -g r` for all extended R CI tests
-* `@github-actions crossbow submit {task-name}` for running a specific task. See the `r:` group definition near the beginning of the [crossbow configuration](https://github.com/apache/arrow/blob/master/dev/tasks/tasks.yml) for a list of glob expression patterns that match names of items in the `tasks:` list below it.
-* `@github-actions autotune` will run and fix lint c++ linting errors + run R documentation (among other cleanup tasks) and commit them to the branch
+#### Run all extended R CI tasks
+`@github-actions crossbow submit -g r`
 
+This runs each of the R-related CI tasks.
 
-## Useful functions for Arrow developers
+#### Run a specific task
+`@github-actions crossbow submit {task-name}`
 
-Within an R session, these can help with package development:
+See the `r:` group definition near the beginning of the [crossbow configuration](https://github.com/apache/arrow/blob/master/dev/tasks/tasks.yml) for a list of glob expression patterns that match names of items in the `tasks:` list below it.
 
-``` r
-# Load the dev package
-devtools::load_all()
+#### Run linting and documentation building tasks
 
-# Run the test suite, optionally filtering file names
-devtools::test(filter="^regexp$")
-# or the Makefile alternative from the arrow/r directory in a shell:
-make test file=regexp
+`@github-actions autotune`
 
-# Update roxygen documentation
-devtools::document()
+This will run and fix lint C++ linting errors, run R documentation (among other cleanup tasks), and commit the resulting updates to the branch.
```

Comment:
```suggestion
This will run and fix lint C++ linting errors, run R documentation (among other cleanup tasks), run styler on any changed R code, and commit the resulting updates to the branch.
```